sql-tuning-best-practice.md

```diff
@@ -0,0 +1,920 @@
+
+## Statistics Management
+
+Statistics are essential to the TiDB optimizer. TiDB uses statistics as input to the optimizer to estimate the number of rows processed in each plan step for a SQL statement.
+
+statistics is generally divided into two levels: table level and column level. 
+
+- For table-level statistics, it includes the total number of rows in the table and the number of rows that have been modified since the last collection of statistics. 
+- The column-level statistics information is more abundant, including histograms, Count-Min Sketch, Top-N (values or indexes with the highest occurrences), distribution and quantity of different values, and the number of null values, and so on.
+
+To ensure the statistics are healthy and representative, you can use the following commands:
+
+1. [`SHOW STATS_META`](sql-statements/sql-statement-show-stats-meta.md): This command provides metadata about table statistics.
+2. [`SHOW STATS_HEALTHY`](sql-statements/sql-statement-show-stats-healthy.md): This command shows the health status of table statistics.
+
+For example, you can use:
+
+```sql
+SHOW STATS_META WHERE table_name='T2'\G;
+```
+
+```
+*************************** 1. row ***************************
+ Db_name: test
+ Table_name: T2
+Partition_name:
+ Update_time: 2023-05-11 02:16:50
+ Modify_count: 20000
+ Row_count: 20000
+1 row in set (0.03 sec)
+```
+
+```sql
+SHOW STATS_HEALTHY WHERE table_name='T2'\G;
+```
+
+```
+*************************** 1. row ***************************
+    Db_name: test
+  Table_name: T2
+Partition_name:
+    Healthy: 0
+1 row in set (0.00 sec)
+```
+
+In TiDB database, there are two ways to collect statistics: automatic collection and manual collection. In most case, the auto collection job works fine. Automatic collection is actually triggered when certain conditions are met for a table, and TiDB will automatically collect statistics. We commonly use three triggering conditions, which are: ratio, start_time and end_time.
+
+- tidb_auto_analyze_ratio: The healthiness trigger
+- [`tidb_auto_analyze_start_time`](/system-variables.md#tidb_auto_analyze_start_time) and [`tidb_auto_analyze_end_time`](/system-variables.md#tidb_auto_analyze_end_time): The allowed job window
+
+```sql
+SHOW VARIABLES LIKE 'tidb\_auto\_analyze%';
+```
+
+```
++-----------------------------------------+-------------+
+| Variable_name                           | Value       |
++-----------------------------------------+-------------+
+| tidb_auto_analyze_ratio                 | 0.5         |
+| tidb_auto_analyze_start_time            | 00:00 +0000 |
+| tidb_auto_analyze_end_time              | 23:59 +0000 |
++-----------------------------------------+-------------+
+```
+
+In cases where automatic collection doesn't meet your needs, you can manually collect statistics using the `ANALYZE TABLE table_name` statement. This allows you to:
```

Comment:
In v8.4, we have enabled the predict analyze. if you miss column's stats, you should run ```analyze table table_name all columns```.

Reply:
@hawkingrei maybe https://docs.pingcap.com/tidb/v8.4/sql-statement-analyze-table should also explain a bit more about the `all columns`?